jujugui/static/gui/src/app/state/state.js

```diff
@@ -202,8 +210,11 @@ const State = class State {
       internal app state. This is typically only done when dispatch is called
       manually. Internal calls to dispatch are done via changeState which
       updates the state itself so it's not needed to do it here.
+    @param {Boolean} backDispatch - Whether this dispatch was from the user
+      hitting the back button. If it was we have to manually figure out the
+      null keys so we know which cleanup dispatchers to run.
   */
-  dispatch(nullKeys = [], updateState = true) {
+  dispatch(nullKeys = [], updateState = true, backDispatch = false) {
```

Comment:
I'm wondering if we are overloading this method by doing this. Maybe we should have a 'backDispatch' method which simply calls dispatch but passing in the `nullkeys` instead of relying on this `backDispatch` argument